iterator.go

```diff
@@ -25,14 +25,29 @@ import (
 //	         return err
 //	     }
 type Iterator struct {
-	c *C.rocksdb_iterator_t
+	c              *C.rocksdb_iterator_t
+	iterUpperBound []byte
```

Comment:
we can reference the `*ReadOptions` directly, slightly simpler and future proof if we'll add new slices later.

Reply:
https://github.com/linxGnu/grocksdb/pull/174

Opened a separate PR to support this practice, WDYT @linxGnu